.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,134 @@
+        run: |
+            # Set up ccache configs .
+            mkdir -p .ccache
+            ccache -sz -M 5Gi
+            
+            # We can not use setup-python as we are running in a container
+            # TODO add to docker image and remove here
+            apt update
+            apt install -y lsb-release python3 pip
+
+      - uses: actions/cache/restore@v3
+        id: restore-cache
+        with:
+          path: ".ccache"
+          key: ccache-benchmark-${{ github.sha }}
+          restore-keys: |
+            ccache-benchmark-
+
+      - name: "Checkout Base"
+        uses: actions/checkout@v3
+        with:
+          path: 'velox'
+          ref: ${{ github.event.pull_request.base.sha || github.sha }}
+          submodules: 'recursive'
+
+      - name: Build Base Benchmarks
+        working-directory: velox
+        run: |
+            n_cores=$(nproc)
+            make benchmarks-basic-build NUM_THREADS=$n_cores MAX_HIGH_MEM_JOBS=$((n_cores/2)) MAX_LINK_JOBS=$n_cores
+            ccache -s
+            mkdir -p ${BINARY_DIR}/baseline/
+            cp -r --verbose _build/release/velox/benchmarks/basic/* ${BINARY_DIR}/baseline/
+
+      - name: "Checkout Head"
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: actions/checkout@v3
+        with:
+          clean: false
+          path: 'velox'
+          submodules: 'recursive'
+
+      - name: Build Target Benchmarks
+        if: ${{ github.event_name == 'pull_request' }}
+        working-directory: velox
+        run: |
+            n_cores=$(nproc)
+            make benchmarks-basic-build NUM_THREADS=$n_cores MAX_HIGH_MEM_JOBS=$n_cores MAX_LINK_JOBS=$n_cores
+            ccache -s
+            mkdir -p  ${BINARY_DIR}/target/
+            cp -r --verbose _build/release/velox/benchmarks/basic/*  ${BINARY_DIR}/target/
+
+      - uses: actions/cache/save@v3
+        id: cache
+        with:
+          path: ".ccache"
+          key: ccache-benchmark-${{ github.sha }}
+
+      - uses: actions/upload-artifact@v3
+        with:
+          path: ${{ env.BINARY_DIR }}
+          name: "benchmark-binaries"
+
+      - name: "Install benchmark dependencies"
+        run: |
+          cd velox
+          python3 -m pip install -r scripts/benchmark-requirements.txt
+
+      - name: "Run Benchmarks - Full Round"
+        working-directory: 'velox'
+        run: |
+            make benchmarks-basic-run \
+                EXTRA_BENCHMARK_FLAGS="--binary_path ${BINARY_DIR}/baseline/ --output_path ${BASELINE_OUTPUT_PATH}"
+
+            if [ "${{ github.event_name }}" == "pull_request" ]; then
+             make benchmarks-basic-run \
+                EXTRA_BENCHMARK_FLAGS="--binary_path ${BINARY_DIR}/target/ --output_path ${TARGET_OUTPUT_PATH}"
+            else
+              echo "Copying baseline results to target for upload."
+              mkdir -p ${TARGET_OUTPUT_PATH}
+              cp -r ${BASELINE_OUTPUT_PATH}/* ${TARGET_OUTPUT_PATH}
+            fi 
```

Comment:
Following on https://github.com/facebookincubator/velox/pull/3830/files#r1089044633 

For example this code wouldn't need to switch on pull request. It cold simply be: run the contender benchmarks, and put those in the right place for upload